action.yml

```diff
@@ -24,23 +24,20 @@ inputs:
     description: "Enabled Nuget Push to Package Registry."
     required: false
     default: "false"
-  nuget_package_name:
-    description: "Creates the nuget package with this name.Used only when nuget_push_enabled is true."
-    required: false
-  nuget_project_path:
-    description: "Path to the Nuget Project File (.csproj).Used only when nuget_push_enabled is true.Required if nuget_push_enabled is true."
-    required: false
   container_push_enabled:
     description: "Enable Build and Push Container Image"
     required: false
     default: "true"
-  github_token:
-    description: "Github token"
+  nuget_pull_token:
+    description: "GitHub token with repo read permissions for pulling NuGet packages"
     required: true
+  nuget_push_token:
+    description: "GitHub token with package write permissions for pushing NuGet packages"
+    required: false
 
 
 runs:
   using: "docker"
-  image: docker://public.ecr.aws/p4v7w0a5/lazy/dotnet-action:v0.1.2
+  image: docker://public.ecr.aws/p4v7w0a5/lazy/dotnet-action:sha-f89e2903df9f056356c9018c99af7774025624da
   env:
-    GITHUB_TOKEN: ${{ inputs.github_token }}
\ No newline at end of file
+    GITHUB_TOKEN: ${{ inputs.nuget_pull_token }}
```

Comment:
Is this tested? I feel that this will mess up normal builds when `dotnet restore` is done. I might be wrong though.

Reply:
Also, if we are using Nuget Token for all dotnet projects, can it be moved ti GitHub runner.

Reply:
@vibindaniel please check out the details in my PR summary. I think my usage in schedule adherence tests this

Reply:
@hpatel826 It works!! Can you keep the token as `github_token` instead of `nuget_pull_token`?

Reply:
@vibindaniel It seems confusing for action users having inputs named github_token and another named nuget_pull_token. It is still set here as GITHUB_TOKEN in the docker env like before. What about nuget_pull_github_token and nuget_push_github_token ? Or maybe even package_pull_github_token and package_push_github_token

Reply:
I thought about this again. And I'm okay with this change

Reply:
I thought about this again, and I'm fine with this change.